### DISCUSSION TITLE
changing if else if statements

### DISCUSSION THREAD

**Discussion by tomaxsas** [tautcony/ISTA-Patcher#38] [General]
I dont have any experience with dnlib. I can make similar patch to already existing ones, but this time need completely different. 
So. How can i change if, else if statements to always evaluate to true or false?
Long method cannot completely rewrite it.

**Comment by tautcony** [comment]
It's a complex topic. The main requirement isn't related to dnlib, but rather knowledge of IL (Intermediate Language) and coming up with an approact to modify instructions to achieve the desired result, instead of simply overwriting them as I do now, after that, use dnlib to implement it. One approach I can think of is replacing with instructions like OpCodes.Jmp, OpCodes.Nop, etc., to skip the code that we don't want to execute.

If you're referring to the function in the Finished Operations patch, I've also had a look on it. On one hand, it's a complex function. To make matters worse, it's an async function, which means during compile, additional state machine related code is generated by compiler. The code that needs to be modified for this function is actually not in the function itself, but in an internal class.

**Comment by tomaxsas** [comment]
Yes, for correctly patching Finished Operations. Will take a look once have time and in mood :D
